### ISSUE TITLE
String::Length is wrong

### ISSUE THREAD

**Issue by zpao** [zpao/v8monkey#17]
We're returning the number of bytes when we should be returning the number of characters while String::Utf8Length returns the number of bytes.

I asked for clarification in case I'm missing something: http://infomonkey.cdleary.com/questions/220/how-can-i-get-the-number-of-actual-characters-from-a-string

**Comment by sdwilsh** [comment]
Can we just divide by two?

**Comment by zpao** [comment]
Is that right though? I think that would make at least one of the tests pass but that could be a contrived example. Well, we'd need to `len / 2 + 1` (for null terminator I guess)

**Comment by zpao** [comment]
Err, no, that's not right. Take this for example:

```
// abc<Icelandic eth><Unicode snowman>.
v8::Handle<String> str2 = v8_str("abc\303\260\342\230\203");
CHECK_EQ(5, str2->Length());
```

We're failing this because we're returning 8 from `Length()`.

**Comment by sdwilsh** [comment]
Oh that's fun.  They are providing that string as UTF-8.  We should be converting that to UTF-16 in the JS engine, and then divide by two should be correct.  However...

**Comment by zpao** [comment]
We want to call JS_SetCStringsAreUTF8, not JS_CStringsAreUTF8. This should start to fix many of the Utf-8 issues I was having, starting with String::Length

Closed by 6611285d74c45e310769375b9f5b1e87bd2282ba
